### ISSUE TITLE
Can SQUID3 process the data from multi-collector mode?

### ISSUE THREAD

**Issue by PENGwxgeo** [CIRDLES/Squid#766]
Squid3 Version: 2.0.5
Java Version: 1.8.0_391
OS: Windows 11 22H2

Can SQUID3 process the data from SHRIMP multi-collector mode? Generally, our lab use the software POXI to process Oxygen/Sulfur isotopes data collected by SHRIMP IIe (multi-collector mode). But I have a Pb isotopes data file and want to process it in SQUID, just in case if there are some U-Pb data collected using multi-collector mode in the future. 
SQUID3 can only recognize only one mass in the xml file. I think it is because there are only one "trim mass" recorded in xml file (see the figure below).

![image](https://github.com/CIRDLES/Squid/assets/66554723/cd57abb1-52ae-4c34-987c-4a61fffb3c06)

The xml file only has one "trim mass", and other masses are recorded as the name of collectors.

**Comment by sbodorkos** [comment]
Hi @PENGwxgeo , unfortunately not. Although we are working to broaden the application of Squid3, it remains confined to single-collector data acquired via ion-counter. POXI remains indispensable for multi-collector data.

Extending Squid3 to multi-collector data would be a major undertaking, and unfortunately neither of the major sponsors of Squid3 (geoscience Australia and the Geological Survey of Canada) has a multi-collector, so it has not been prioritised.

**Comment by PENGwxgeo** [comment]
That's fine. The data mentioned before is acquired in 2021 and processed using Excel with custom formulas. I just borrowed this data file from someone and try to compare the differences between single-collector xml and multi-collector xml, If all the algorithms that SQUID3 built-in (like SBM normalization, dead-time correction, or/and error propagation etc.) are visible in the project, I can dig out these algorithms and process the multi-collector data manually for now. Because I think our custom formulas made some mistakes in calculating errors. I just found a few algorithms in "squidCore" folder. Are there any other algorithms in other folder(s)? 
Maybe in the future, I can upload some codes to enable SQUID to process multi-collector data, but I am not sure because I just started trying to learn and understand the algorithms used in SQUID (earlier verison and 3) and the software for the other SIMS company. It will take a long time I think.

**Comment by bowring** [comment]
I suggest you read all of https://github.com/CIRDLES/Squid/wiki for an understanding of the data reduction processes - the algorithms you seek are described in detail therein.

**Comment by PENGwxgeo** [comment]
> I suggest you read all of https://github.com/CIRDLES/Squid/wiki for an understanding of the data reduction processes - the algorithms you seek are described in detail therein.

Thank you very much!!

**Comment by sbodorkos** [comment]
@PENGwxgeo Following through that wiki will fundamentally lead you to this point:

https://github.com/CIRDLES/ET_Redux/wiki/SHRIMP:-Step-3

which is where SQUID performs Dodson interpolation of single-collector data to generate time-equivalent values for numerator and denominator in the isotopic ratios. In theory, multi-collector data ought to permit simplifications here, in cases of genuinely simultaneous data acquisition. But coding it might be complicated...

**Comment by PENGwxgeo** [comment]
> @PENGwxgeo Following through that wiki will fundamentally lead you to this point:
> 
> https://github.com/CIRDLES/ET_Redux/wiki/SHRIMP:-Step-3
> 
> which is where SQUID performs Dodson interpolation of single-collector data to generate time-equivalent values for numerator and denominator in the isotopic ratios. In theory, multi-collector data ought to permit simplifications here, in cases of genuinely simultaneous data acquisition. But coding it might be complicated...

Yes, both SQUID and the other SIMS software use statistical or experiential method to try to make the counts and ratios closer to the "true" value. It needs careful calculation .....
